core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -319,6 +319,7 @@ public synchronized void writeHeaders(Metadata headers) {
 
       @Override
       public void close(Status status, Metadata trailers) {
+        status = stripCause(status);
```

Comment:
This is `InProcessServerStream.close(...)`, which invokes `clientStreamListener.closed`. That would be like a RST_STREAM, therefore stripped.